provisioning/device/samples/register_x509.js

```diff
@@ -1,34 +1,78 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
-
 'use strict';
+var X509Security = require('azure-iot-security-x509').X509Security;
+var ProvisioningDeviceClient = require('azure-iot-provisioning-device').ProvisioningDeviceClient;
 
 var fs = require('fs');
-var Transport = require('azure-iot-provisioning-device-http').Http;
+var argv = require('yargs')
+  .usage('Usage: $0  --idscope <ID SCOPE> --registrationid <REGISTRATION ID> --certificate <COMMONNAME> --protocol <PROTOCOL>')
+  .option('idscope', {
+    alias: 'i',
+    describe: 'The ID scope is assigned to a Device Provisioning Service',
+    type: 'string',
+    demandOption: true
+  })
+  .option('registrationid', {
+    alias: 'r',
+    describe: 'Used to uniquely identify a device in the Device Provisioning Servic',
+    type: 'string',
+    demandOption: true
+  })
+  .option('certificate', {
+    alias: 'c',
+    describe: 'Commonname of the certificate without extension. Expected filenames: commonName_cert.pem and commonName_key.pem ',
```

Comment:
>commonName_cert.pem and commonName_key.pem [](start = 84, length = 42)

i'm not sure about the value of this convention over having simply 2 parameters.